components/bounty-detail/bounty-detail-client.tsx

```
@@ -81,6 +84,23 @@ export function BountyDetailClient({ bountyId }: { bountyId: string }) {
+  const competitionSubmissions = (
+    (bounty as { submissions?: Array<{
+      id: string;
+      submittedBy: string;
+      submittedByUser?: { name?: string | null; image?: string | null } | null;
+      githubPullRequestUrl?: string | null;
+      status: string;
+    }> }).submissions ?? []
+  );
+  const pastDeadline =
+    bounty.bountyWindow?.endDate != null &&
+    Date.now() > new Date(bounty.bountyWindow.endDate).getTime();
```

ESLint

[error] Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/components/bounty-detail/bounty-detail-client.tsx:102:5
  100 |   const pastDeadline =
  101 |     bounty.bountyWindow?.endDate != null &&
> 102 |     Date.now() > new Date(bounty.bountyWindow.endDate).getTime();
      |     ^^^^^^^^^^ Cannot call impure function
  103 |
  104 |   return (
  105 |     <div className="flex flex-col lg:flex-row gap-10">

(react-hooks/purity)